src/app/services/push.service.ts

```diff
@@ -60,65 +51,30 @@ export class PushService {
     }
   }
 
-  private initPush() {
-    PushService.pushObject = this.push.init({
-      android: {},
-      ios: {
-        alert: true,
-        badge: true,
-        sound: true,
-      },
+  public register() {
+    new PushRegistration(new ConfigurationService(config))
+    .register({
+      alias: 'cordova',
+      categories: ['ionic', 'showcase']
+    }).then(() => {
+      PushService.registered = true;
+      console.log('Push registration successful');
+    }).catch((err) => {
+      PushService.registered = false;
+      console.error('Push registration unsuccessful ', err);
     });
   }
 
-  private emit(notification: PushMessage) {
-    if (PushService.callback) {
-      PushService.callback(notification);
-    }
-  }
-
-  // No longer receive notifications
   public unregister() {
-    PushService.pushObject.unregister().then(() => {
+    new PushRegistration(new ConfigurationService(config))
+    .unregister()
+    .then(() => {
       PushService.registered = false;
       console.log('Successfully unregistered');
-    }).catch(() => {
-      console.error('Error unregistering');
-    });
-  }
-
-  public register() {
-    PushService.pushObject.on('error').subscribe((err) => {
-      this.pushError = err;
-      console.error(`Error configuring push notifications ${err.message}`);
-    });
-
-    // Invokes the UPS registration endpoint
-    PushService.pushObject.on('registration').subscribe((data) => {
-      new PushRegistration(new ConfigurationService(config)).register(data.registrationId, PUSH_ALIAS, ['ionic', 'showcase']).then(() => {
-        PushService.registered = true;
-        console.log('Push registration successful');
-      }).catch((err) => {
-        this.pushError = err;
-        console.error('Push registration unsuccessful ', this.pushError);
-      });
-    });
-
-    PushService.pushObject.on('notification').subscribe((notification) => {
-      const newNotification = {
-        title: notification.title,
-        message: notification.message,
-        received: new Date().toDateString(),
-      };
-      this.emit(newNotification);
+    }).catch((err) => {
+      PushService.registered = true;
+      console.error('Error unregistering', err);
     });
   }
 
-  public getError() {
-    return this.pushError;
-  }
-
-  public isRegistered() {
```

Comment:
If we remove this then we probably do not need to have ugly static registered field

Reply:
Removed!